Content.Server/GameObjects/Components/NodeContainer/Nodes/PipeNode.cs

```diff
@@ -111,13 +109,26 @@ protected override IEnumerable<Node> GetReachableNodes()
                 {
                     continue;
                 }
-                var pipeNodesInDirection = Owner.GetComponent<SnapGridComponent>()
-                    .GetInDir(pipeDirection.ToDirection())
-                    .Select(entity => entity.TryGetComponent<NodeContainerComponent>(out var container) ? container : null)
-                    .Where(container => container != null)
-                    .SelectMany(container => container.Nodes)
-                    .OfType<PipeNode>()
-                    .Where(pipeNode => pipeNode._pipeDirection.HasFlag(theirNeededConnection));
+
+                var pipeNodesInDirection = new List<PipeNode>();
+
+                var entities = Owner.GetComponent<SnapGridComponent>()
+                    .GetInDir(pipeDirection.ToDirection());
+
+                foreach (var entity in entities)
+                {
+                    if (entity.TryGetComponent<NodeContainerComponent>(out var container))
+                    {
+                        foreach (var node in container.Nodes)
+                        {
+                            if (node is PipeNode pipeNode && pipeNode._pipeDirection.HasFlag(theirNeededConnection))
```

Comment:
As a heads up, `HasFlag` is very slow! You might want to create an extension method for this, check how AtmosDirection does it.

Reply:
It's just for debug afaik but this still.